sympy/printing/str.py

```diff
@@ -74,13 +74,16 @@ def _print_BooleanTrue(self, expr):
     def _print_BooleanFalse(self, expr):
         return "False"
 
+    def _print_Not(self, expr):
+        return '~(%s)' %(expr.args[0])
+
     def _print_And(self, expr):
-        return '%s(%s)' % (expr.func, ', '.join(sorted(self._print(a) for a in
-            expr.args)))
+        return '%s' %(' & '.join([self.parenthesize(arg, 38)
```

Comment:
Well, it depends on what you are printing. Different targets have different precedence.

Let's make it simple, let's just use any trick to solve this (i.e. arbitrary value for precedence) and then write a short comment about why you did so.